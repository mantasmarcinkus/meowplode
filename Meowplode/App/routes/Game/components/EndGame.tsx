import { Path } from "history";
import * as React from "react";
import { Router } from "react-router";
import { AppStore } from "../../../store/appStore";
import Leaderboard from "../../Leaderboard/container";
import { ActionCreators, DefinedState } from "../module";
const KeyHandler = (require("react-key-handler") as any).default;
const KEYPRESS = (require("react-key-handler") as any).KEYPRESS;
const KEYUP = (require("react-key-handler") as any).KEYUP;

const TEXT = (success: boolean) => success
  ? {
    buttonText: "Let's save another one!",
    newPlayerText: "New saver?",
    title: "Yay, you saved a kitten",
  }
  : {
    buttonText: "Try to save atleast one",
    newPlayerText: "Let someone else try..",
    title: "So sad...",
  };

interface Props extends Pick<DefinedState & ActionCreators,
  "name" | "clearGame" | "startGame" | "success"> {
  router: Path[] | Router;
}

export default class EndGame extends React.Component<Props, {}> {
  constructor() {
    super();
    this.onRetryClick = this.onRetryClick.bind(this);
    this.onNewPlayerClick = this.onNewPlayerClick.bind(this);
  }
  public render() {
    const texts = TEXT(this.props.success);
    return (
      <div>
        <KeyHandler keyEventName={KEYPRESS} keyValue={"Enter"} onKeyHandle={this.onRetryClick} />
        <KeyHandler keyEventName={KEYUP} keyValue={"Escape"} onKeyHandle={this.onNewPlayerClick} />
        <div style={{ fontSize: 28, margin: 30 }}>{texts.title}</div>
        <button className="border" onClick={this.onRetryClick}>{texts.buttonText}</button><br />
        <button className="answer" onClick={this.onNewPlayerClick}>{texts.newPlayerText}</button>
        <div style={{ marginTop: 20 }}>
          <Leaderboard name={this.props.name} />
        </div>
        <div style={{ fontSize: 10, marginTop: 15 }}>
          <a href="https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK">
            Find out more about kittens! (Hints here!)
          </a>
        </div>
      </div>
    );
  }
  private onRetryClick() {
    this.props.startGame(this.props.name);
  }
  private onNewPlayerClick() {
    this.props.clearGame();
    (this.props.router as Path[]).push("/");
  }
}
