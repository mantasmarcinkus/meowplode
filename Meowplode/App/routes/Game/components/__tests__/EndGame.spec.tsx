import { shallow, ShallowWrapper } from "enzyme";
import * as React from "react";
import { Dispatch } from "redux";
import EndGame from "../EndGame";

jest.mock("../../../Leaderboard/container");

let startGameResult: string = "";
let clearGameResult: boolean = false;

const testProps = {
  clearGame: () => { clearGameResult = true; },
  name: "text",
  router: [] as string[],
  startGame: (name: string) => (dispatch: Dispatch<any>) => { startGameResult = name; },
  success: true,
};

describe("<EndGame />", () => {
  let wrapper: ShallowWrapper<any, any>;
  beforeEach(() => {
    wrapper = shallow(<EndGame {...testProps} />);
  });
  it("renders", () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
  it("onRetryClick calls startGame", () => {
    (wrapper.instance() as any).onRetryClick();
    expect(startGameResult).toEqual("");
  });
  it("onRetryClick calls clear game", () => {
    (wrapper.instance() as any).onNewPlayerClick();
    expect(clearGameResult).toEqual(true);
  });
});
