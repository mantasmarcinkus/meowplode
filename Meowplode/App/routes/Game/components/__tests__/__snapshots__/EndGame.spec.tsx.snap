// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EndGame /> renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "style": "font-size:28px;margin:30px;",
            },
            "children": Array [
              Object {
                "data": "Yay, you saved a kitten",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "next": Object {
              "attribs": Object {
                "class": "border",
              },
              "children": Array [
                Object {
                  "data": "Let's save another one!",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "next": Object {
                "attribs": Object {},
                "children": Array [],
                "name": "br",
                "next": Object {
                  "attribs": Object {
                    "class": "answer",
                  },
                  "children": Array [
                    Object {
                      "data": "New saver?",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": Object {
                    "attribs": Object {
                      "style": "margin-top:20px;",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "name": "text",
                        },
                        "children": Array [],
                        "name": "test-file-stub",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": Object {
                      "attribs": Object {
                        "style": "font-size:10px;margin-top:15px;",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "href": "https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK",
                          },
                          "children": Array [
                            Object {
                              "data": "Find out more about kittens! (Hints here!)",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                      ],
                      "name": "div",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "class": "border",
            },
            "children": Array [
              Object {
                "data": "Let's save another one!",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "next": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "br",
              "next": Object {
                "attribs": Object {
                  "class": "answer",
                },
                "children": Array [
                  Object {
                    "data": "New saver?",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": Object {
                  "attribs": Object {
                    "style": "margin-top:20px;",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "name": "text",
                      },
                      "children": Array [],
                      "name": "test-file-stub",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": Object {
                    "attribs": Object {
                      "style": "font-size:10px;margin-top:15px;",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "href": "https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK",
                        },
                        "children": Array [
                          Object {
                            "data": "Find out more about kittens! (Hints here!)",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "a",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "style": "font-size:28px;margin:30px;",
              },
              "children": Array [
                Object {
                  "data": "Yay, you saved a kitten",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "br",
            "next": Object {
              "attribs": Object {
                "class": "answer",
              },
              "children": Array [
                Object {
                  "data": "New saver?",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "next": Object {
                "attribs": Object {
                  "style": "margin-top:20px;",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "name": "text",
                    },
                    "children": Array [],
                    "name": "test-file-stub",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": Object {
                  "attribs": Object {
                    "style": "font-size:10px;margin-top:15px;",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "href": "https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK",
                      },
                      "children": Array [
                        Object {
                          "data": "Find out more about kittens! (Hints here!)",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "a",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "border",
              },
              "children": Array [
                Object {
                  "data": "Let's save another one!",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "style": "font-size:28px;margin:30px;",
                },
                "children": Array [
                  Object {
                    "data": "Yay, you saved a kitten",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "class": "answer",
            },
            "children": Array [
              Object {
                "data": "New saver?",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "next": Object {
              "attribs": Object {
                "style": "margin-top:20px;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "name": "text",
                  },
                  "children": Array [],
                  "name": "test-file-stub",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
              ],
              "name": "div",
              "next": Object {
                "attribs": Object {
                  "style": "font-size:10px;margin-top:15px;",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "href": "https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK",
                    },
                    "children": Array [
                      Object {
                        "data": "Find out more about kittens! (Hints here!)",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "br",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "border",
                },
                "children": Array [
                  Object {
                    "data": "Let's save another one!",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "style": "font-size:28px;margin:30px;",
                  },
                  "children": Array [
                    Object {
                      "data": "Yay, you saved a kitten",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "style": "margin-top:20px;",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "name": "text",
                },
                "children": Array [],
                "name": "test-file-stub",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
            ],
            "name": "div",
            "next": Object {
              "attribs": Object {
                "style": "font-size:10px;margin-top:15px;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK",
                  },
                  "children": Array [
                    Object {
                      "data": "Find out more about kittens! (Hints here!)",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
              ],
              "name": "div",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "answer",
              },
              "children": Array [
                Object {
                  "data": "New saver?",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [],
                "name": "br",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "border",
                  },
                  "children": Array [
                    Object {
                      "data": "Let's save another one!",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "style": "font-size:28px;margin:30px;",
                    },
                    "children": Array [
                      Object {
                        "data": "Yay, you saved a kitten",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "tag",
                },
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "style": "font-size:10px;margin-top:15px;",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "https://www.buzzfeed.com/chelseamarshall/meows?utm_term=.clGnJmJZ4#.nvAZ6z6lK",
                },
                "children": Array [
                  Object {
                    "data": "Find out more about kittens! (Hints here!)",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
            ],
            "name": "div",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "style": "margin-top:20px;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "name": "text",
                  },
                  "children": Array [],
                  "name": "test-file-stub",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
              ],
              "name": "div",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "answer",
                },
                "children": Array [
                  Object {
                    "data": "New saver?",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [],
                  "name": "br",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "border",
                    },
                    "children": Array [
                      Object {
                        "data": "Let's save another one!",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "style": "font-size:28px;margin:30px;",
                      },
                      "children": Array [
                        Object {
                          "data": "Yay, you saved a kitten",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                  "type": "tag",
                },
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
